pkg/controllers/user/helm/controller.go

```diff
@@ -347,9 +374,23 @@ func (l *Lifecycle) createAppRevision(obj *v3.App, template, notes string, faile
 func (l *Lifecycle) writeKubeConfig(obj *v3.App, tempDir string, remove bool) (string, error) {
 	var token string
 
-	userID := obj.Annotations["field.cattle.io/creatorId"]
+	isSystemProject, err := l.isInSystemProject(obj)
+	if err != nil {
+		return "", err
+	}
+
+	userID, ok := obj.Annotations["field.cattle.io/creatorId"]
 	user, err := l.UserClient.Get(userID, metav1.GetOptions{})
-	if err != nil && !errors.IsNotFound(err) {
+	if !ok && isSystemProject {
```

Comment:
the different conditions of this set of if/else statements are very hard to understand. you should add comments to explain